### DISCUSSION TITLE
Python/Pip: can't override WORKON_HOME env variable and prevents readonly errors in /home/ubuntu/.local/share/virtualenvs

### DISCUSSION THREAD

**Discussion by jgarec** [renovatebot/renovate#26806] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.142.1-full docker image

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

Hello, we are running renovate-ce in a kubernetes cluster.
The container is executed in readonly mode except the tmp folder which is mounted with R/W privilegies

This issue is also reproductible with the full docker image executed with read-only flag :

```
docker run --read-only --rm -it -v ${PWD}/tmp:/tmp -e RENOVATE_TOKEN renovate/renovate:37.142.1-full --platform=gitlab jgarec/poc-renovate-python
```

The issue is reproductible with this repository : https://gitlab.com/jgarec/poc-renovate-python/-/merge_requests/2 
and shows that there is write accesses in the home folder.

I'm not a python expert but i think with WORKON_HOME set to /tmp/something this should be ok but there is no way to use this parameter. Maybe because this parameter is not handled here : 
https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/pipenv/artifacts.ts#L141

### Relevant debug logs

<details><summary>Logs</summary>

```
  File "/opt/containerbase/tools/pipenv/2023.11.17/3.12/lib/python3.12/site-packages/pipenv/utils/shell.py", line 195, in get_workon_home
    os.makedirs(expanded_path, exist_ok=True)
  File "<frozen os>", line 225, in makedirs
OSError: [Errno 30] Read-only file system: '/home/ubuntu/.local/share/virtualenvs'

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
PRs welcome to add this env to allowed default env

**Comment by viceice** [comment]
you can use custom environment variables

https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables

**Reply by jgarec** [reply]
oh thx, i didn't know this paremeter.
Yes it works but there is no cleanup as it's not handled as a cache directory. So a PR is still necessary ?

**Reply by viceice** [reply]
not sure, what's in that folder?

**Reply by viceice** [reply]
why do you need that be removed after run?

I don't think we will set it out of the box. renovate is not supported to run on a read-only file system. it may work, but renovate depends on containerbase for installing tools on the fly. which definitely isn't working on read-only file system. 

@rarkins do you think we should set `WORKON_HOME` inside pipenv manager to the renovate cache dir? 🤔

**Reply by rarkins** [reply]
Yes, we could set it by default

**Comment by jgarec** [comment, marked as answer]
Fixed in 37.149.0 with #26818
